src/sentry/models/release.py

```diff
@@ -129,14 +130,42 @@ def filter_to_semver(self):
         """
         return self.filter(major__isnull=False)
 
+    def filter_by_semver_build(
+        self, organization_id: int, operator: str, build: str, project_ids: Sequence[int] = None
+    ) -> models.QuerySet:
+        """
+        Filters released by build. If the passed `build` is a numeric string, we'll filter on
+        `build_number` and make use of the passed operator.
+        If it is a non-numeric string, then we'll filter on `build_code` instead. We support a
+        wildcard only at the end of this string, so that we can filter efficiently via the index.
+        """
+        qs = self.filter(organization_id=organization_id)
+
+        if project_ids:
+            qs = qs.filter(
+                id__in=ReleaseProject.objects.filter(project_id__in=project_ids).values_list(
+                    "release_id", flat=True
+                )
+            )
+
+        if build.isnumeric():
+            qs = qs.filter(**{f"build_number__{operator}": int(build)})
```

Comment:
Why are we not checking if build is >63 bits here

Reply:
🤦  I went to all the trouble to move `validate_bigint` to a utility function then forgot to call it, thanks!